### PR TITLE
Refactor `urns` package

### PR DESCRIPTION
**Pull request by rowanseymour**
Trying to...

 1. Fix https://github.com/nyaruka/gocommon/issues/85
 2. Simplify the API (don't have a bunch of type specific things)
 3. For phone numbers.. only do conversion to an international number at parse time and not in `Normalize` because only at parse time do we typically have the right country code.
 4. Enforce max length of 255 across all schemes

### CODE REVIEW

File: urns/urns.go [hunk 1]
```diff
@@ -3,125 +3,33 @@ package urns
...
-
-// NewWhatsAppURN returns a URN for the passed in whatsapp identifier
-func NewWhatsAppURN(identifier string) (URN, error) {
-	return NewURNFromParts(WhatsAppScheme, identifier, "", "")
-}
-
-// NewFirebaseURN returns a URN for the passed in firebase identifier
-func NewFirebaseURN(identifier string) (URN, error) {
-	return NewURNFromParts(FCMScheme, identifier, "", "")
-}
```
Comment (rowanseymour) [inline urns/urns.go:102]: we only created constructors for some schemes.. and we don't need them

Comment (norkans7) [reply urns/urns.go:102]: Those were used in courier but we can update courier to use `NewURNFromParts`

Comment (norkans7) [reply urns/urns.go:102]: Or as it is most common to not have a `query` and `display` for the URN can we have a short cut that we pass the `scheme` type and the `path` and that set the `query` and `display` to `""`?

Comment (rowanseymour) [reply urns/urns.go:102]: Have added that as `urns.New(scheme, path)`

**Review by norkans7 — APPROVED** [review]
Looks good, we could also update the go version to 1.22

**Pull request merged**